utils.js

```diff
@@ -83,6 +83,81 @@ const GlobalSignalsHandler = new Lang.Class({
 });
 
 /**
+ * Color manipulation utilities
+ * 
+ * Most of this code comes from reposts on stackoverflow, I was unable to trace the
+ * original authors, otherwise I would have credited them here.
+ */
+const ColorUtils = new Lang.Class({
+    Name: 'DashToDock.ColorUtils',
+    
+    ColorLuminance: function(r, g, b, lum) {
+        var hex = b | (g << 8) | (r << 16);
+        
+        hex = (0x1000000 + hex).toString(16).slice(1);
+
+        // convert to decimal and change luminosity
+        var rgb = "#", c, i;
+        for (i = 0; i < 3; i++) {
+            c = parseInt(hex.substr(i*2,2), 16);
+            c = Math.round(Math.min(Math.max(0, c + (c * lum)), 255)).toString(16);
+            rgb += ("00"+c).substr(c.length);
+        }
+
+        return rgb;
+    },
+
+    HSVtoRGB: function(h, s, v) {
+        var r, g, b, i, f, p, q, t;
+        if (arguments.length === 1) {
+            s = h.s, v = h.v, h = h.h;
+        }
+        i = Math.floor(h * 6);
+        f = h * 6 - i;
+        p = v * (1 - s);
+        q = v * (1 - f * s);
+        t = v * (1 - (1 - f) * s);
+        switch (i % 6) {
+            case 0: r = v, g = t, b = p; break;
+            case 1: r = q, g = v, b = p; break;
+            case 2: r = p, g = v, b = t; break;
+            case 3: r = p, g = q, b = v; break;
+            case 4: r = t, g = p, b = v; break;
+            case 5: r = v, g = p, b = q; break;
+        }
+        return {
+            r: Math.round(r * 255),
+            g: Math.round(g * 255),
+            b: Math.round(b * 255)
+        };
+    },
+
+    RGBtoHSV: function (r, g, b) {
+        if (arguments.length === 1) {
+            g = r.g, b = r.b, r = r.r;
+        }
+        var max = Math.max(r, g, b), min = Math.min(r, g, b),
+            d = max - min,
+            h,
+            s = (max === 0 ? 0 : d / max),
+            v = max / 255;
+
+        switch (max) {
+            case min: h = 0; break;
+            case r: h = (g - b) + d * (g < b ? 6: 0); h /= 6 * d; break;
+            case g: h = (b - r) + d * 2; h /= 6 * d; break;
+            case b: h = (r - g) + d * 4; h /= 6 * d; break;
+        }
+
+        return {
+            h: h,
+            s: s,
+            v: v
+        };
+    }
+});
+
+/**
```

Comment:
Hi Michele,
Thank you for your quick and enthusiastic response! I am happy that there is interest in my changes, I will investigate if I can use the HLS native code (since its not the same as HSV)